test/core/event_engine/test_suite/tests/dns_test.cc

```diff
@@ -25,7 +68,747 @@ void InitDNSTests() {}
 }  // namespace grpc_event_engine
 
+#ifdef GPR_WINDOWS
 class EventEngineDNSTest : public EventEngineTest {};
 
-// TODO(hork): establish meaningful tests
+// TODO(yijiem): make the test run on Windows
 TEST_F(EventEngineDNSTest, TODO) { grpc_core::ExecCtx exec_ctx; }
+#else
+
+namespace {
+
+using grpc_event_engine::experimental::EventEngine;
+using SRVRecord = EventEngine::DNSResolver::SRVRecord;
+using testing::ElementsAre;
+using testing::Pointwise;
+using testing::UnorderedPointwise;
+
+// TODO(yijiem): make this portable for Windows
+constexpr char kDNSTestRecordGroupsYamlPath[] =
+    "test/core/event_engine/test_suite/tests/dns_test_record_groups.yaml";
+constexpr char kHealthCheckRecordName[] =
+    "health-check-local-dns-server-is-alive.resolver-tests.grpctestingexp";
+
+MATCHER(ResolvedAddressEq, "") {
+  const auto& addr0 = std::get<0>(arg);
+  const auto& addr1 = std::get<1>(arg);
+  return addr0.size() == addr1.size() &&
+         memcmp(addr0.address(), addr1.address(), addr0.size()) == 0;
+}
+
+MATCHER(SRVRecordEq, "") {
+  const auto& arg0 = std::get<0>(arg);
+  const auto& arg1 = std::get<1>(arg);
+  return arg0.host == arg1.host && arg0.port == arg1.port &&
+         arg0.priority == arg1.priority && arg0.weight == arg1.weight;
+}
+
+MATCHER(StatusCodeEq, "") {
+  return std::get<0>(arg).code() == std::get<1>(arg);
+}
+
+// Copied from tcp_socket_utils_test.cc
+// TODO(yijiem): maybe move those into common test util
+EventEngine::ResolvedAddress MakeAddr4(const uint8_t* data, size_t data_len,
+                                       int port) {
+  EventEngine::ResolvedAddress resolved_addr4;
+  sockaddr_in* addr4 = reinterpret_cast<sockaddr_in*>(
+      const_cast<sockaddr*>(resolved_addr4.address()));
+  memset(&resolved_addr4, 0, sizeof(resolved_addr4));
+  addr4->sin_family = AF_INET;
+  GPR_ASSERT(data_len == sizeof(addr4->sin_addr.s_addr));
+  memcpy(&addr4->sin_addr.s_addr, data, data_len);
+  addr4->sin_port = htons(port);
+  return EventEngine::ResolvedAddress(
+      reinterpret_cast<sockaddr*>(addr4),
+      static_cast<socklen_t>(sizeof(sockaddr_in)));
+}
+
+EventEngine::ResolvedAddress MakeAddr6(const uint8_t* data, size_t data_len,
+                                       int port) {
+  EventEngine::ResolvedAddress resolved_addr6;
+  sockaddr_in6* addr6 = reinterpret_cast<sockaddr_in6*>(
+      const_cast<sockaddr*>(resolved_addr6.address()));
+  memset(&resolved_addr6, 0, sizeof(resolved_addr6));
+  addr6->sin6_family = AF_INET6;
+  GPR_ASSERT(data_len == sizeof(addr6->sin6_addr.s6_addr));
+  memcpy(&addr6->sin6_addr.s6_addr, data, data_len);
+  addr6->sin6_port = htons(port);
+  return EventEngine::ResolvedAddress(
+      reinterpret_cast<sockaddr*>(addr6),
+      static_cast<socklen_t>(sizeof(sockaddr_in6)));
+}
+
+#define EXPECT_UNKNOWN_ERROR(result) \
+  EXPECT_EQ((result).status().code(), absl::StatusCode::kUnknown)
+
+}  // namespace
+
+class EventEngineDNSTest : public EventEngineTest {
+ public:
```

Comment:
Everything here can be `protected`.  There's no need for the `SetUpTestSuite()` and `TearDownTestSuite()` methods to be public.

Reply:
Done.